src/backend/TrafficCourts/Citizen.Service/Features/Disputes/Create.cs

```diff
@@ -41,56 +42,42 @@ public Response(Exception exception)
-                    OcrViolationTicket violationTicket = await _redisCacheService.GetRecordAsync<OcrViolationTicket>(ocrKey);
-                    ocrViolationTicketJson = JsonSerializer.Serialize(violationTicket);
+                    OcrViolationTicket? violationTicket = await _redisCacheService.GetRecordAsync<OcrViolationTicket>(ocrKey);
+                    if (violationTicket != null)
+                    {
+                        ocrViolationTicketJson = JsonSerializer.Serialize(violationTicket);
+                    }
                 }
+
+                SubmitNoticeOfDispute submitNoticeOfDispute = _mapper.Map<SubmitNoticeOfDispute>(createDisputeRequest);
+                submitNoticeOfDispute.ViolationTicket.OcrViolationTicket = ocrViolationTicketJson;
+                submitNoticeOfDispute.CitizenSubmittedDate = DateTime.UtcNow;
                 
-                var response = await _submitDisputeRequestClient.GetResponse<DisputeSubmitted>(new
-                {
-                    TicketNumber = createDisputeRequest.TicketNumber,
-                    CourtLocation = createDisputeRequest.CourtLocation,
-                    ViolationDate = createDisputeRequest.ViolationDate,
-                    GivenNames = createDisputeRequest.GivenNames,
-                    DisputantSurname = createDisputeRequest.DisputantSurname,
-                    StreetAddress = createDisputeRequest.StreetAddress,
-                    Province = createDisputeRequest.Province,
-                    PostalCode = createDisputeRequest.PostalCode,
-                    HomePhone = createDisputeRequest.HomePhone,
-                    EmailAddress = createDisputeRequest.EmailAddress,
-                    DriversLicence = createDisputeRequest.DriversLicence,
-                    DriversLicenceProvince = createDisputeRequest.DriversLicenceProvince,
-                    WorkPhone = createDisputeRequest.WorkPhone,
-                    DateOfBirth = createDisputeRequest.DateOfBirth,
-                    EnforcementOrganization = createDisputeRequest.EnforcementOrganization,
-                    ServiceDate = createDisputeRequest.ServiceDate,
-                    TicketCounts = createDisputeRequest.TicketCounts,
-                    LawyerRepresentation = createDisputeRequest.LawyerRepresentation,
-                    InterpreterLanguage = createDisputeRequest.InterpreterLanguage,
-                    WitnessIntent = createDisputeRequest.WitnessIntent,
-                    OcrViolationTicket = ocrViolationTicketJson
-                });
+                var response = await _submitDisputeRequestClient.GetResponse<DisputeSubmitted>(submitNoticeOfDispute);
 
                 // Send email message to the submitter's entered email
                 var template = MailTemplateCollection.DefaultMailTemplateCollection.FirstOrDefault(t => t.TemplateName == "SubmitDisputeTemplate");
```

Comment:
I dont think sending email should be here. We should be handling the `SubmitNoticeOfDispute` event in a handler and have it send email

Reply:
I haven't made this change and it's already in main so I think it would be better to address this on a separate PR in order to unblock this one.

Reply:
Agreed, lets address this in a separate PR.

Reply:
lets make a jira ticket for this or update an existing one